google-cloud-core/src/main/java/com/google/cloud/HttpTransportOptions.java

```diff
@@ -154,10 +157,29 @@ public void initialize(HttpRequest httpRequest) throws IOException {
         if (readTimeout >= 0) {
           httpRequest.setReadTimeout(readTimeout);
         }
+
+        HttpHeaders headers = httpRequest.getHeaders();
+        headers.set("x-goog-api-client", xApiClientHeader());
       }
     };
   }
 
+  @VisibleForTesting
+  String xApiClientHeader() {
+    return String.format(
+        "gl-java/%s %s/%s gax/%s grpc/%s",
```

Comment:
This format is actually different from the ones used in GAPIC (GAPIC header has one extra 'gapic' field)